exasol/bucketfs/bucket_path.py

```diff
@@ -0,0 +1,231 @@
+from __future__ import annotations
+from typing import ByteString, BinaryIO, Iterable, Optional, Generator
+from pathlib import PurePath, PureWindowsPath
+import errno
+import os
+from io import IOBase
+from exasol.bucketfs.bucket_api import BucketApi
+from exasol.bucketfs._pathlike import Pathlike
+
+
+class _BucketFile:
+    """
+    A node in a perceived file structure of a bucket.
+    This can be a file, a directory or both.
+    """
+
+    def __init__(self, name: str, parent: str = ''):
+        self._name = name
+        self._path = f'{parent}/{name}' if parent else name
+        self._children: Optional[dict[str, "_BucketFile"]] = None
+        self.is_file = False
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def path(self):
+        return self._path
+
+    @property
+    def is_dir(self):
+        # The node can be a directory as well as a file,
+        # hence is the is_dir property, independent of is_file.
+        return bool(self._children)
+
+    def __iter__(self):
+        if self._children is None:
+            return iter(())
+        return iter(self._children.values())
+
+    def get_child(self, child_name: str) -> "_BucketFile":
```

Comment:
consider naming function which do not need to be "public" with a leading `_`, e.g: `_get_child`

Reply:
It's a public method